include/seastar/core/shared_ptr.hh

```diff
@@ -608,12 +633,15 @@ public:
     explicit operator bool() const noexcept {
         return _p;
     }
+    [[clang::callable_when("unconsumed", "unknown")]]
     T& operator*() const noexcept {
         return *_p;
     }
+    [[clang::callable_when("unconsumed", "unknown")]]
     T* operator->() const noexcept {
         return _p;
     }
+    [[clang::callable_when("unconsumed", "unknown")]]
```

Comment:
Why is "unknown" ok?

Reply:
it is based on practical reasons. the state of an consumable object is based on annotations, but not all the code paths are annotated. in the code paths where the annotation is missing, and compiler is not able to deduce the state of the object, it's state would be "unknown", but we should not fail the build just because the compiler is not able to figure out the runtime behavior without enough annotations. actually, we have lots of them in scylla.

to make the `-Wconsume` useful before annotating all of them, adding "unknown" is a good compromise. probably is a must.